From 7f92869f4e3e9f5694994362dab228b55fae8735 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 5 Sep 2014 14:53:09 +0800
Subject: [PATCH 0570/5242] MLK-11703-5: ARM: dts: imx6: add battery device
 node in dts file

commit  c18af6b57b81a86e24ead05137610c681497055f from
https://source.codeaurora.org/external/imx/linux-imx.git

cherry-pick below patch:

ENGR00330406-1: ARM: dts: imx6: add battery device node in dts file

Add battery device node on imx6q/dl-sabresd and imx6sl-evk board

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit 1f2670578decd4a527208864ab3a014d840773f6)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dl-sabresd.dts   |    6 ++++++
 arch/arm/boot/dts/imx6q-sabresd.dts    |    6 ++++++
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi |   13 +++++++++++++
 arch/arm/boot/dts/imx6sl-evk.dts       |   16 ++++++++++++++++
 4 files changed, 41 insertions(+)

diff --git a/arch/arm/boot/dts/imx6dl-sabresd.dts b/arch/arm/boot/dts/imx6dl-sabresd.dts
index b87747b..c8103ff5 100644
--- a/arch/arm/boot/dts/imx6dl-sabresd.dts
+++ b/arch/arm/boot/dts/imx6dl-sabresd.dts
@@ -12,6 +12,12 @@
 	compatible = "fsl,imx6dl-sabresd", "fsl,imx6dl";
 };
 
+&battery {
+	offset-charger = <1485>;
+	offset-discharger = <1464>;
+	offset-usb-charger = <1285>;
+};
+
 &iomuxc {
         epdc {
                 pinctrl_epdc_0: epdcgrp-0 {
diff --git a/arch/arm/boot/dts/imx6q-sabresd.dts b/arch/arm/boot/dts/imx6q-sabresd.dts
index 4ac66f2..e08c79d 100644
--- a/arch/arm/boot/dts/imx6q-sabresd.dts
+++ b/arch/arm/boot/dts/imx6q-sabresd.dts
@@ -13,6 +13,12 @@
 	compatible = "fsl,imx6q-sabresd", "fsl,imx6q";
 };
 
+&battery {
+	offset-charger = <1900>;
+	offset-discharger = <1694>;
+	offset-usb-charger = <1685>;
+};
+
 &ldb {
 	lvds-channel@0 {
 		crtc = "ipu2-di0";
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 62732e9..cc20c62 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -15,6 +15,19 @@
 		mxcfb3 = &mxcfb4;
 	};
 
+	battery: max8903@0 {
+		compatible = "fsl,max8903-charger";
+		pinctrl-names = "default";
+		dok_input = <&gpio2 24 1>;
+		uok_input = <&gpio1 27 1>;
+		chg_input = <&gpio3 23 1>;
+		flt_input = <&gpio5 2 1>;
+		fsl,dcm_always_high;
+		fsl,dc_valid;
+		fsl,usb_valid;
+		status = "okay";
+	};
+
 	hannstar_cabc {
 		compatible = "hannstar,cabc";
 		lvds0 {
diff --git a/arch/arm/boot/dts/imx6sl-evk.dts b/arch/arm/boot/dts/imx6sl-evk.dts
index 6825173..74ae571 100644
--- a/arch/arm/boot/dts/imx6sl-evk.dts
+++ b/arch/arm/boot/dts/imx6sl-evk.dts
@@ -12,6 +12,19 @@
 	model = "Freescale i.MX6 SoloLite EVK Board";
 	compatible = "fsl,imx6sl-evk", "fsl,imx6sl";
 
+	battery: max8903@0 {
+		compatible = "fsl,max8903-charger";
+		pinctrl-names = "default";
+		dok_input = <&gpio4 13 1>;
+		uok_input = <&gpio4 13 1>;
+		chg_input = <&gpio4 15 1>;
+		flt_input = <&gpio4 14 1>;
+		fsl,dcm_always_high;
+		fsl,dc_valid;
+		fsl,adc_disable;
+		status = "okay";
+	};
+
 	memory@80000000 {
 		reg = <0x80000000 0x40000000>;
 	};
@@ -463,6 +476,9 @@
 				MX6SL_PAD_EPDC_PWRCTRL2__GPIO2_IO09 0x80000000
 				MX6SL_PAD_KEY_COL6__GPIO4_IO04    0x110b0
 				MX6SL_PAD_LCD_RESET__GPIO2_IO19   0x1b0b0
+				MX6SL_PAD_ECSPI2_MISO__GPIO4_IO14 0x17000
+				MX6SL_PAD_ECSPI2_MOSI__GPIO4_IO13 0x17000
+				MX6SL_PAD_ECSPI2_SS0__GPIO4_IO15  0x17000
 			>;
 		};
 
-- 
1.7.9.5

